Updated models to sync with ServiceHub team [#158037410]

YouKnow232 opened the pull request.

YouKnow232 committed:
Updated models to sync with ServiceHub team [#158037410]

YouKnow232 committed:
Refactoring model references to be inline with the new models [#158037410]

YouKnow232 commented:
Directly copying model implementation from the service hub team is proving to be problematic.  This can be seen especially in the unit tests.

For instance:
A User contain a Name object, which itself contains a list of Users objects with this Name, each User in this list contains the aforementioned Name object etc.

JGrippo reviewed (changes requested):
Remove the nav properties from the child models.

YouKnow232 committed:
Commented out circular dependencies [#158037410]

Only the User object has references to batch and room.
Address and Name no longer have references to their containing object.
These changes are made with comments as they are subject to change.

YouKnow232 committed:
Refactored unit tests to work with new models [#158037410]

JGrippo reviewed (approved):
Looks good.
